compiler/rustc_middle/src/mir/mono.rs

```diff
@@ -103,17 +102,14 @@ impl<'tcx> MonoItem<'tcx> {
                 // inlined function. If we're inlining into all CGUs then we'll
                 // be creating a local copy per CGU.
                 if generate_cgu_internal_copies {
-                    return InstantiationMode::LocalCopy;
-                }
-
-                // Finally, if this is `#[inline(always)]` we're sure to respect
-                // that with an inline copy per CGU, but otherwise we'll be
-                // creating one copy of this `#[inline]` function which may
-                // conflict with upstream crates as it could be an exported
-                // symbol.
-                match tcx.codegen_fn_attrs(instance.def_id()).inline {
-                    InlineAttr::Always => InstantiationMode::LocalCopy,
-                    _ => InstantiationMode::GloballyShared { may_conflict: true },
+                    InstantiationMode::LocalCopy
+                } else {
+                    // Finally, if this is `#[inline(always)]` or `#[inline]` and
+                    // OptLevel is not No, we're sure to respect that with an inline
+                    // copy per CGU, but otherwise we'll be creating one copy of
+                    // this `#[inline]` function which may conflict with upstream
+                    // crates as it could be an exported symbol.
```

Comment:
Nit: Suggested rewrite for this comment. Please tweak as you see fit :) 

```suggestion
                    // Finally, if we've reached this point, then we know that OptLevel = No
                    // so we should optimize for compilation speed. In that regard, we will
                    // ignore any `#[inline]` annotations on the function and simply codegen
                    // it as usual. This could conflict with upstream crates as it could be an exported
                    // symbol.
```